splice_machine/src/main/java/com/splicemachine/derby/ddl/DDLUtils.java

```diff
@@ -382,6 +377,16 @@ public static void preDropSchema(DDLMessage.DDLChange change, DataDictionary dd,
         dd.getDataDictionaryCache().schemaCacheRemove(change.getDropSchema().getSchemaName());
     }
 
+    public static void preUpdateSchemaOwner(DDLMessage.DDLChange change, DataDictionary dd, DependencyManager dm) throws StandardException {
+        if (LOG.isDebugEnabled())
+            SpliceLogUtils.debug(LOG,"preUpdateSchemaOwner with change=%s",change);
+        dd.getDataDictionaryCache().schemaCacheRemove(change.getUpdateSchemaOwner().getSchemaName());
+        // clear permission cache as it has out-of-date permission info for the schema
+        dd.getDataDictionaryCache().clearPermissionCache();
+        // clear  TableDescriptor cache as it may reference the schema with an out-of-date authorization id
+        dd.getDataDictionaryCache().clearOidTdCache();
```

Comment:
Doesn't the normal dm.invalidateFor() work here?

Reply:
@dgomezferro that does not invalidate permission cache.

Reply:
@yxia92 Does schema cache and nameTdCache need to be invalidated?

Reply:
@jyuanca Thanks for the comment. Yes, nameTdCache needs to be invalidated too. For schemaCache, I think we have invalidated the specific item in DDLUtils.preUpdateSchemaOwner() with the code 
{code}      dd.getDataDictionaryCache().schemaCacheRemove(change.getUpdateSchemaOwner().getSchemaName());
{code}
Is it not enough?

Reply:
I've added code to invalidate nameTdCache too.

Reply:
@yxia92 I missed that part :-)